src/java.rs

```diff
@@ -124,16 +109,16 @@ pub fn deobfuscate_bytecode_signature(
 }
 
 /// formats types (param_type list, return_type) into a human-readable signature
-pub fn format_signature(types: &Option<(Vec<String>, String)>) -> Option<String> {
-    if types.is_none() {
-        return None;
-    }
+pub fn format_signature(types: Option<DeobfuscatedSignature>) -> Option<String> {
+    let types = types?;
```

Comment:
This as well should delegate the `Option`s to the caller.

It might also be a good idea to move this method onto the `DeobfuscatedSignature` struct.
Either as a separate method, or as the `Display` implementation. WDYT @loewenheim ?

Reply:
I agree, this would make a nice `Display` impl.